src/status_im/chat/models/input.cljs

```diff
@@ -239,23 +240,33 @@
                                 :text         (i18n/label :t/update-to-see-sticker {"locale" "en"})})))
 
 (rf/defn send-edited-message
-  [{:keys [db] :as cofx} text {:keys [message-id quoted-message chat-id]}]
-  (rf/merge
-   cofx
-   {:json-rpc/call [{:method      "wakuext_editMessage"
-                     :params      [{:id           message-id
-                                    :text         text
-                                    :content-type (if (message-content/emoji-only-content?
-                                                       {:text text :response-to quoted-message})
-                                                    constants/content-type-emoji
-                                                    constants/content-type-text)}]
-                     :js-response true
-                     :on-error    #(log/error "failed to edit message " %)
-                     :on-success  (fn [result]
-                                    (re-frame/dispatch [:sanitize-messages-and-process-response
-                                                        result]))}]}
-   (link-preview/reset-unfurled)
-   (cancel-message-edit)))
+  [{:keys [db]
+    :as   cofx} text {:keys [message-id quoted-message chat-id]}]
+  (let [rename-data-uri-fn #(update %
+                                    :thumbnail
+                                    (fn [thumbnail]
+                                      (set/rename-keys thumbnail {:data-uri :dataUri})))]
+    (rf/merge
+     cofx
+     {:json-rpc/call [{:method      "wakuext_editMessage"
+                       :params      [{:id            message-id
+                                      :text          text
+                                      :content-type  (if (message-content/emoji-only-content?
+                                                          {:text        text
+                                                           :response-to quoted-message})
+                                                       constants/content-type-emoji
+                                                       constants/content-type-text)
+                                      :link-previews (map #(-> %
+                                                               (select-keys [:url :title :description :thumbnail])
+                                                               rename-data-uri-fn)
```

Comment:
This code is duplicating logic that we can find in https://github.com/status-im/status-mobile/blob/255a3b917265d381fed57e26422277b4cc8772e4/src/status_im/data_store/messages.cljs#L18 and https://github.com/status-im/status-mobile/blob/255a3b917265d381fed57e26422277b4cc8772e4/src/status_im/data_store/messages.cljs#L55

Also note that the we can skip the anonymous function when using `update` with `rename-keys`. I forgot this myself when I wrote the function `<-link-preview-rpc`.

```clojure
;; Unnecessary anonymous fn
(defn- <-link-preview-rpc
  [preview]
  (update preview
          :thumbnail
          (fn [thumbnail]
            (set/rename-keys thumbnail {:dataUri :data-uri}))))

;; Better
(defn- <-link-preview-rpc
  [preview]
  (update preview :thumbnail set/rename-keys {:dataUri :data-uri}))

(<-link-preview-rpc {:thumbnail {:dataUri "https://example.xyz"}})
;; => {:thumbnail {:data-uri "https://example.xyz"}}
```

Reply:
Done, Thanks Icaro!